pkg/controller/certificates/certificate_controller_test.go

```diff
@@ -64,7 +64,8 @@ func TestCertificateController(t *testing.T) {
 
 	stopCh := make(chan struct{})
 	defer close(stopCh)
-	go informerFactory.Start(stopCh)
+	informerFactory.Start(stopCh)
+	informerFactory.WaitForCacheSync(stopCh)
```

Comment:
controller.processNextWorkItem() is called synchronously in this test. Are you talking about interactions within the informer? I have separated the controller and the informer client so that I can only assert the interaction the processNextWorkItem should take.